reporter/gcppubsub/pubsub_test.go

```diff
@@ -0,0 +1,99 @@
+package gcppubsub
+
+import (
+	"context"
+	"fmt"
+	"github.com/openzipkin/zipkin-go/model"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+var topicID string
+
+var once sync.Once // guards cleanup related operations in setup.
+
+func setup() *pubsub.Client {
+	ctx := context.Background()
+	proj := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	fmt.Printf("GCP Project: %s\n", proj)
+	topicID = "test-topic"
+
+	client, err := pubsub.NewClient(ctx, proj)
+	if err != nil {
+		fmt.Printf("failed to create client: %s\n", topicID)
+		return nil
+	}
+
+	_, err = client.CreateTopic(ctx, topicID)
+	if err != nil {
+		fmt.Printf("failed to create topic: %v", err)
+		return nil
+	}
+	fmt.Printf("Topic created: %s\n", topicID)
+	return client
+}
+
+func TestPublish(t *testing.T) {
```

Comment:
maybe worth to run this test with two test cases: one with topic and another one with the default topic?